src/overlays/actors/ovl_En_Kakasi3/z_en_kakasi3.c

```diff
@@ -21,35 +47,382 @@ const ActorInit En_Kakasi3_InitVars = {
+        this->dialogState = 5;
+        func_8002DF54(globalCtx, NULL, 8);
+        this->actionFunc = func_80A91A90;
+    } else if (globalCtx->msgCtx.unk_E3EE == 1) {
+        func_80A90EBC(this, globalCtx, 0);
+        player->stateFlags2 |= 0x800000;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi3/func_80A91620.s")
+void func_80A91A90(EnKakasi3* this, GlobalContext* globalCtx) {
+    func_80A90E28(this);
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    func_8002DF54(globalCtx, NULL, 8);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi3/func_80A91760.s")
+    if (this->dialogState == func_8010BDBC(&globalCtx->msgCtx) && func_80106BC8(globalCtx)) {
+        if (this->unk_195 != 0) {
+            if ((gSaveContext.eventChkInf[9] & 0x1000) == 0) {
+                gSaveContext.eventChkInf[9] |= 0x1000;
+            }
+        }
+        if (globalCtx->cameraPtrs[this->camId] == 0) {
+            this->camId = -1;
+        }
+        if (this->camId != -1) {
+            func_8005B1A4(globalCtx->cameraPtrs[this->camId]);
+        }
+        func_80106CCC(globalCtx);
+        globalCtx->msgCtx.unk_E3EE = 4U;
+        func_8002DF54(globalCtx, NULL, 7);
+        this->actionFunc = func_80A911F0;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi3/func_80A917FC.s")
+void EnKakasi3_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnKakasi3* this = THIS;
+    s32 pad;
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi3/func_80A9187C.s")
+    if (BREG(2) != 0) {
+        osSyncPrintf("\n\n");
+        // flag!
+        osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ フラグ！ ☆☆☆☆☆ %d\n" VT_RST, gSaveContext.unk_12C5);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi3/func_80A918E4.s")
+    this->unk_198 += 1;
+    this->actor.posRot.rot = this->actor.shape.rot;
+    for (i = 0; i < 4; i++) {
+        if (this->unk_19C[i] != 0) {
+            this->unk_19C[i]--;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi3/func_80A91A90.s")
+    Actor_SetHeight(&this->actor, 60.0f);
+    this->actionFunc(this, globalCtx);
+    Actor_MoveForward(&this->actor);
+    func_8002E4B4(globalCtx, &this->actor, 50.0f, 50.0f, 100.0f, 28);
+    Collider_CylinderUpdate(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi3/EnKakasi3_Update.s")
+void EnKakasi3_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnKakasi3* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi3/EnKakasi3_Draw.s")
+    func_80093D18(globalCtx->state.gfxCtx);
+    SkelAnime_DrawSV(globalCtx, this->skelAnime.skeleton, this->skelAnime.limbDrawTbl, this->skelAnime.dListCount, 0, 0,
+                     &this->actor);
```

Comment:
```suggestion
    SkelAnime_DrawSV(globalCtx, this->skelAnime.skeleton, this->skelAnime.limbDrawTbl, this->skelAnime.dListCount, NULL, NULL,
                     &this->actor);
```
pass `NULL` for the function pointers callbacks